radish/parser.py

```diff
@@ -436,33 +453,45 @@ def _parse_skip_scenario(self, line):
 
         return True
 
+    def _detect_keyword(self, keyword, line):
+        """
+        Detects a keyword on a given line
+        :param keyword: the keyword to detect
+        :param line: the line in which we want to detect the keyword
+        :return: the line without the detected keyword
+        :rtype: string or None
+        """
+        pattern = "^%s\s*%s" % (keyword, self._keywords_delimiter)
+        match = re.match(pattern, line)
+
+        if match:
+            return line[line.index(self._keywords_delimiter) + 1:].strip()
```

Comment:
Well, no - you still format the string with the `keyword` and `keyword_delimiter` but instead of finding the index of the keyword delimiter in case of a match and using it for a string slice to get everything after the delimiter you just use the group which was matched. Something along the lines of:

```python
pattern = "^%s\s*%s(.*)$" % (keyword, self._keywords_delimiter)
match = re.match(pattern, line)

if match:
    return match.group(1).strip()

return None
```